[0.78.1] Release build failure

### Environment

```markdown
react-native -v: 0.78.1
npm ls react-native-macos: 0.78.1
node -v: v20.14.0
npm -v: 10.9.2
yarn --version: n/a
xcodebuild -version: Xcode 16.1, Build version 16B40
```

### Steps to reproduce the bug

1. Initialize a fresh project with 0.78.1
2. Runs fine in dev
3. Errors in release

### Expected Behavior

Build should succeed

### Actual Behavior

Build failed with these warnings/errors

```
In file included from /Users/jay/Documents/code/test-projects/rnmac78/node_modules/react-native-macos/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTInstance.mm:12:
In file included from /Users/jay/Documents/code/test-projects/rnmac78/macos/Pods/Headers/Public/React-RCTFBReactNativeSpec/FBReactNativeSpec/FBReactNativeSpec.h:24:
/Users/jay/Documents/code/test-projects/rnmac78/macos/Pods/Headers/Public/RCTTypeSafety/RCTTypeSafety/RCTConvertHelpers.h:27:24: warning: use of GNU ?: conditional expression extension,
      omitting middle operand [-Wgnu-conditional-omitted-operand]
   27 |     array[i] = object ?: (id)kCFNull;
      |                        ^
/Users/jay/Documents/code/test-projects/rnmac78/node_modules/react-native-macos/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTInstance.mm:213:1: warning: implementing deprecated
      method [-Wdeprecated-implementations]
  213 | - (NSArray<id<RCTBridgeModule>> *)extraModulesForBridge:(RCTBridge *)bridge
      | ^
In file included from /Users/jay/Documents/code/test-projects/rnmac78/node_modules/react-native-macos/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTInstance.mm:36:
/Users/jay/Documents/code/test-projects/rnmac78/macos/Pods/Headers/Public/React-NativeModulesApple/ReactCommon/RCTTurboModuleManager.h:53:1: note: method 'extraModulesForBridge:' declared
      here
   53 | - (NSArray<id<RCTBridgeModule>> *)extraModulesForBridge:(RCTBridge *)bridge
      | ^
/Users/jay/Documents/code/test-projects/rnmac78/node_modules/react-native-macos/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTInstance.mm:216:29: warning: 'extraModulesForBridge:'
      is deprecated: Please make all native modules returned from this method TurboModule-compatible. [-Wdeprecated-declarations]
  216 |     return [_appTMMDelegate extraModulesForBridge:nil];
      |                             ^
In file included from /Users/jay/Documents/code/test-projects/rnmac78/node_modules/react-native-macos/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTInstance.mm:36:
/Users/jay/Documents/code/test-projects/rnmac78/macos/Pods/Headers/Public/React-NativeModulesApple/ReactCommon/RCTTurboModuleManager.h:54:18: note: 'extraModulesForBridge:' has been
      explicitly marked deprecated here
   54 |     __attribute((deprecated("Please make all native modules returned from this method TurboModule-compatible.")));
      |                  ^
/Users/jay/Documents/code/test-projects/rnmac78/node_modules/react-native-macos/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTInstance.mm:520:19: warning: 
      'instance:didReceiveJSErrorStack:message:originalMessage:name:componentStack:exceptionId:isFatal:extraData:' is deprecated [-Wdeprecated-declarations]
  520 |   if (![_delegate instance:self
      |                   ^
In file included from /Users/jay/Documents/code/test-projects/rnmac78/node_modules/react-native-macos/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTInstance.mm:8:
/Users/jay/Documents/code/test-projects/rnmac78/node_modules/react-native-macos/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTInstance.h:56:85: note: 
      'instance:didReceiveJSErrorStack:message:originalMessage:name:componentStack:exceptionId:isFatal:extraData:' has been explicitly marked deprecated here
   56 |                  extraData:(NSDictionary<NSString *, id> *)extraData __attribute__((deprecated));
      |                                                                                     ^
/Users/jay/Documents/code/test-projects/rnmac78/node_modules/react-native-macos/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTInstance.mm:527:34: error: cannot initialize a
      parameter of type 'BOOL' (aka 'signed char') with an rvalue of type 'id _Nullable'
  527 |                          isFatal:errorData[@"isFatal"]
      |                                  ^~~~~~~~~~~~~~~~~~~~~
In file included from /Users/jay/Documents/code/test-projects/rnmac78/node_modules/react-native-macos/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTInstance.mm:8:
/Users/jay/Documents/code/test-projects/rnmac78/node_modules/react-native-macos/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTInstance.h:55:34: note: passing argument to parameter
      'isFatal' here
   55 |                    isFatal:(BOOL)isFatal
      |                                  ^
4 warnings and 1 error generated.
```

### Reproducible Demo

_No response_

### Additional context

_No response_

--- a/packages/react-native/Libraries/Text/TextInput/RCTBaseTextInputView.mm
+++ b/packages/react-native/Libraries/Text/TextInput/RCTBaseTextInputView.mm
@@ -513,6 +513,13 @@ RCT_NOT_IMPLEMENTED(-(instancetype)initWithCoder : (NSCoder *)decoder)
 
 - (void)textInputDidBeginEditing
 {
+#if TARGET_OS_OSX // [macOS
+  if (_isCurrentlyEditing) {
+    return;
+  }
+  _isCurrentlyEditing = YES;
+#endif // macOS]
+
   [_eventDispatcher sendTextEventWithType:RCTTextEventTypeFocus
                                  reactTag:self.reactTag
                                      text:[self.backedTextInputView.attributedText.string copy]

--- a/packages/react-native/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTInstance.mm
+++ b/packages/react-native/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTInstance.mm
@@ -524,7 +524,7 @@ void RCTInstanceSetRuntimeDiagnosticFlags(NSString *flags)
                             name:errorData[@"name"]
                   componentStack:errorData[@"componentStack"]
                      exceptionId:error.id
-                         isFatal:errorData[@"isFatal"]
+                         isFatal:[errorData[@"isFatal"] boolValue]
                        extraData:errorData[@"extraData"]]) {
     JS::NativeExceptionsManager::ExceptionData jsErrorData{errorData};
     id<NativeExceptionsManagerSpec> exceptionsManager = [_turboModuleManager moduleForName:"ExceptionsManager"];

--- a/packages/react-native/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTInstance.mm
+++ b/packages/react-native/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTInstance.mm
@@ -524,7 +524,11 @@ void RCTInstanceSetRuntimeDiagnosticFlags(NSString *flags)
                             name:errorData[@"name"]
                   componentStack:errorData[@"componentStack"]
                      exceptionId:error.id
-                         isFatal:[errorData[@"isFatal"] boolValue]
+#if TARGET_OS_OSX // [macOS Use boolValue for isFatal to ensure type safety
+                     isFatal:[errorData[@"isFatal"] boolValue] // Explicitly convert to BOOL to avoid compiler errors on macOS
+#else
+                     isFatal:errorData[@"isFatal"]
+#endif // macOS]
                        extraData:errorData[@"extraData"]]) {
     JS::NativeExceptionsManager::ExceptionData jsErrorData{errorData};
     id<NativeExceptionsManagerSpec> exceptionsManager = [_turboModuleManager moduleForName:"ExceptionsManager"];

